simulation/amaru-sim/src/simulator/generate.rs

```diff
@@ -329,12 +319,89 @@ pub fn generate_arrival_times<R: Rng>(
     arrival_times
 }
 
+pub fn generate_entries<R: Rng>(
+    file_path: &PathBuf,
+    start_time: Instant,
+    mean_millis: f64,
+    number_of_upstream_peers: u8,
+) -> impl Fn(&mut R) -> Vec<Reverse<Entry<ChainSyncMessage>>> {
```

Comment:
@stevana you had added `+ use<'_, R>` here and so I went and investigated that new feature; it seems that `+ '_` would have done the trick, but then I thought why not update to edition 2024 which does this by default, so we'll need to fix this anyway. Feel free to remove my commit from your branch again if you don’t like it.